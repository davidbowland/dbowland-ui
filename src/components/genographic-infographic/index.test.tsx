import React from 'react'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import gatsyPluginImage from 'gatsby-plugin-image'

import GenographicInfographic from './index'

jest.mock('gatsby-plugin-image')

describe('Genographic infographic component', () => {
  beforeAll(() => {
    jest.spyOn(gatsyPluginImage, 'StaticImage').mockImplementation((props) => {
      return <img {...props} />
    })
  })

  test('Ensure snapshot match', () => {
    const { container } = render(<GenographicInfographic />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('Ensure image is on page', () => {
    const { getByRole } = render(<GenographicInfographic />)

    const img = getByRole('img') as HTMLImageElement
    expect(img.src).toContain('genographic-infographic.png')
  })
})
