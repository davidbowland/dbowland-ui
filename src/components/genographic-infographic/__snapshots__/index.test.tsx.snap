// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genographic infographic component Ensure snapshot match 1`] = `
<figure
  class="sc-bdvvaa aTyKP"
>
  <a
    href="test-file-stub"
  >
    <img
      alt="41% Eastern Europe; 22% Western and Central Europe; 20% Scandinavin; 14% Great Britain and Ireland; 2% Asia Minor; 1.1% Neanderthal"
      src="../../assets/images/genographic-infographic.png"
    />
  </a>
  <figcaption
    class="sc-gsDJrp eIPapg"
  >
    Click or tap on the image for the full 18-page PDF report.
  </figcaption>
</figure>
`;
